flexget/utils/radarr.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import unicode_literals, division, absolute_import
+from builtins import *
+from future.moves.urllib.parse import urlparse, quote
+from future.utils import python_2_unicode_compatible
+
+import logging
+import json
+import requests
+from requests import RequestException
+
+log = logging.getLogger('radarr')
+
+
+@python_2_unicode_compatible
+class RadarrRequestError(Exception):
+    def __init__(self, value, logger=log, **kwargs):
+        super(RadarrRequestError, self).__init__()
+        # Value is expected to be a string
+        if not isinstance(value, str):
+            value = str(value)
+        self.value = value
+        self.log = logger
+        self.kwargs = kwargs
+
+    def __str__(self):
+        return self.value
+
+class RadarrMovieAlreadyExistsError(Exception):
+    def __init__(self):
+        super(RadarrMovieAlreadyExistsError, self).__init__()
+
+def spec_exception_from_response_ex(radarr_request_ex):
+
+    status_code = None
+    error_message = None
+
+    if 'status_code' in radarr_request_ex.kwargs:
+        status_code = radarr_request_ex.kwargs['status_code']
+
+    if 'error_message' in radarr_request_ex.kwargs:
+        error_message = radarr_request_ex.kwargs['error_message']
+
+    if not error_message:
+        return None
+
+    if error_message.lower() == 'this movie has already been added':
+        return RadarrMovieAlreadyExistsError()
+
+    return None
+
+def request_get_json(url, headers):
+    """ Makes a GET request and returns the JSON response """
+    try:
+        response = requests.get(url, headers=headers)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise RadarrRequestError('Invalid response received from Radarr: %s' % response.content)
+    except RequestException as e:
+        raise RadarrRequestError('Unable to connect to Radarr at %s. Error: %s' % (url, e))
+
+def request_delete_json(url, headers):
+    """ Makes a DELETE request and returns the JSON response """
+    try:
+        response = requests.delete(url, headers=headers)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            raise RadarrRequestError('Invalid response received from Radarr: %s' % response.content)
+    except RequestException as e:
+        raise RadarrRequestError('Unable to connect to Radarr at %s. Error: %s' % (url, e))
+
+def request_post_json(url, headers, data):
+    """ Makes a POST request and returns the JSON response """
+    try:
+        response = requests.post(url, headers=headers, data=data)
+        if response.status_code == 201:
+            return response.json()
+        else:
+            errorMessage = None
+            try:
+                json_response = response.json()
+                if len(json_response) > 0:
+                    errorMessage = json_response[0]['errorMessage']
+            except:
```

Comment:
Never do this... At the very minimum add some logs here, but try to avoid the generic except / pass paradigm like an STD...